### ISSUE TITLE
RevEng: Not generting FK properties

### ISSUE THREAD

**Issue by rowanmiller** [aspnet/EntityFramework#1938]
```
CREATE TABLE [dbo].[Category](
	[CategoryId] [int] IDENTITY(1,1) NOT NULL,
	[DisplayName] [nvarchar](max) NULL,
	CONSTRAINT [PK_Category] PRIMARY KEY NONCLUSTERED ( [CategoryId] ASC ))
GO

CREATE TABLE [dbo].[Product](
	[ProductId] [int] IDENTITY(1,1) NOT NULL,
	[CurrentPrice] [decimal](18, 2) NOT NULL,
	[Description] [nvarchar](max) NULL,
	[CategoryId] [int] NOT NULL,
 CONSTRAINT [PK_Product] PRIMARY KEY NONCLUSTERED  ( [ProductId] ASC ))

ALTER TABLE [dbo].[Product]  WITH CHECK ADD  CONSTRAINT [FK_Product_Category_CategoryId] FOREIGN KEY([CategoryId])
REFERENCES [dbo].[Category] ([CategoryId])
GO
```

Note there is no `CategoryId` property in Product class:

```
    public partial class Product
    {
        public Product()
        {
            CartItem = new HashSet<CartItem>();
            OrderLine = new HashSet<OrderLine>();
        }
        
        // Properties
        public int ProductId { get; set; }
        public decimal CurrentPrice { get; set; }
        public string Description { get; set; }
        
        // Navigation Properties
        public virtual Category Category { get; set; }
    }
```

**Comment by lajones** [comment]
Note: where the name of the NavProp would match the name of the property (e.g. see #1076) how would you like the name of the NavProp calculated - since the name of the property will no longer be available?

**Comment by rowanmiller** [comment]
What are out current rules... do we just remove 'Id' from the end of the FK column name to get the navigation name?

**Comment by lajones** [comment]
The name of the Nav prop pointing from the principal to the dependent end is always just the name of the dependent end's EntityType (uniquifed by adding 1, 2... if there is a clash).

For the name of the Nav prop pointing from the dependent end to the principal the current rules are:
1) take all the properties representing the FK and find the common prefix (e.g. if we have 2 props called 'PrincipalId1' and 'PrincipalId2' we'd get 'PrincipalId')
2) strip off any trailing 'Id', 'ID' or any casing of '_id'
3) if what's left is non-empty go to step 5
4) if what's left is empty (e.g. the 2 FK properties don't have a common prefix) then we use the principal entity type's name
5) uniquify the result to prevent clashes as necessary.

**Comment by lajones** [comment]
Note: an example where we would get a clash between FK property name and Nav Prop name: on Employees in Northwind there is a self-association via a column called 'ReportsTo'. Per #1076 we generate a NavProp called 'ReportsTo'. But at the moment we don't include the FK property itself in the model. If we did include a 'ReportsTo' property then what should the NavProp be called? The current rules would call it 'ReportsTo1' which is better than the old 'Employees1' but would you prefer something different?

**Comment by rowanmiller** [comment]
Discussed in triage, we will append 'Navigation' to the end of the navigation property name when needed

**Comment by lajones** [comment]
Fixed by PR #1968
